src/CMakeLists.txt

```diff
@@ -183,6 +179,8 @@ IF (USE_HTTPS)
 		LIST(APPEND LIBGIT2_LIBS ${COREFOUNDATION_LIBRARIES} ${SECURITY_LIBRARIES})
 		LIST(APPEND LIBGIT2_PC_LIBS ${COREFOUNDATION_LDFLAGS} ${SECURITY_LDFLAGS})
 	ELSEIF (HTTPS_BACKEND STREQUAL "OpenSSL")
+		FIND_PACKAGE(OpenSSL)
```

Comment:
My original intention was to have all the various "Is this available" tests outside of the main `USE_HTTPS` block (or at least at the beginning of it). The `mbedTLS` backend might be used as some kind of reference. I don't really care in any way, just stating what intentions where (and at the time, I refrained from hoisting that `FIND_PACKAGE` because of `AMIGA`).

Reply:
Dunno. Personally, I find it preferential to keep packages as close as possible to the location where they're actually used. Furthermore, your original implementation has the problem that you've added a `QUIET` afterwards -- so it can never be seen that we're actually checking for OpenSSL. On the other hand, if we remove that `QUIET`, then it's always being checked regardless of whether it is actually requested.

That's why, at least in my opinion, the change improves the situation.

Reply:
I have no hard opinion about it, since doing the check isn't that expensive, and might actually help pinpointing a non-detection event of OpenSSL (it has a helpful "pass `OPENSSL_ROOT_DIR`, which `QUIET` swallows). In the end, it's up to the `HTTPS` block to actually decide whether to use the found package or not.